#!/usr/bin/env python
PACKAGE="masterslave"

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()
jointWeights = gen.add_group("JointWeights")
jointWeights.add("joint1", double_t, 0, "The joint weight for joint 1", 1, 0, 1000)
jointWeights.add("joint2", double_t, 0, "The joint weight for joint 2", 1, 0, 1000)
jointWeights.add("joint3", double_t, 0, "The joint weight for joint 3", 1, 0, 1000)
jointWeights.add("joint4", double_t, 0, "The joint weight for joint 4", 1, 0, 1000)
jointWeights.add("joint5", double_t, 0, "The joint weight for joint 5", 1, 0, 1000)
jointWeights.add("joint6", double_t, 0, "The joint weight for joint 6", 1, 0, 1000)
jointWeights.add("joint7", double_t, 0, "The joint weight for joint 7", 1, 0, 1000)
jointWeights.add("joint8", double_t, 0, "The joint weight for joint 8", 1, 0, 1000)
jointWeights.add("joint9", double_t, 0, "The joint weight for joint 9", 1, 0, 1000)
jointWeights.add("joint10", double_t, 0, "The joint weight for joint 10", 1, 0, 1000)

gen.add("AccelerationGain", double_t, 0, "AccelerationGain", 0.0000001, 0, 1)
gen.add("VelocityGain", double_t, 0, "VelocityGain", 0.0000001, 0, 1)
gen.add("AngleMaximumGain", double_t, 0, "AngleMaximumGain", 0.0000001, 0, 1)
gen.add("CollisionAvoidanceGain", double_t, 0, "CollisionAvoidanceGain", 0.0000001, 0, 1)
gen.add("SingularityAvoidanceGain", double_t,0, "SingulariryAvoidanceGain", 0.0000001, 0, 1);
gen.add("TrocarGain", double_t, 0, "TrocarGaiN", 1, 0, 1)
gen.add("TCPGain", double_t, 0, "TCPGain", 1, 0, 1) 
gen.add("MaxSpeed", double_t, 0, "maxJointSpeedoftheLBR",1, 0, 2)
exit(gen.generate(PACKAGE, "NumericKinematic", "NumericKinematic"))
